### PR TITLE
Add giftshop to urban city block and 3 new items

### PR DESCRIPTION
**Pull request by snipercup**
#### Summary
Content "Add gift shop to urban city block plus three new items"

#### Purpose of change

More content, helps #34049

#### Describe the solution

Adds a giftshop to the urban city block. While the layout is arbitrary, the items that are sold are based on a real-life giftshop in Salem, which is in new england. Except I added some items that aren't sold in the real-life giftshop.

Adds 3 new items:
- figurine
- malachite
- incense

#### Describe alternatives you've considered

Going full realism mode and only add what's actually being sold in this specific giftshop

#### Testing

Started a game and visited the giftshop

#### Additional context

image

### CODE REVIEW

**Comment by ashraf1246** [comment]
شو اخبارك

File: data/json/items/generic/toys_and_sports.json [hunk 1]
```diff
@@ -53,6 +53,33 @@
...
+    "symbol": "f",
+    "color": "yellow",
+    "name": { "str": "figurine" },
+    "description": "Some kind of figurine.  It does not have much use nowadays.",
+    "snippet_category": [
+      { "id": "genfig01", "text": "Shaped like a human female.  It was made to wear different clothes." },
+      { "id": "genfig02", "text": "This one wears a hat.  It wears a glove on it's right hand that has three blades sticking out." },
+      { "id": "genfig03", "text": "A blue unicorn that covers it's eyes with it's hooves.  It's in a sitting position." },
+      { "id": "genfig04", "text": "A pink unicorn that covers it's ears with it's hooves.  It's in a sitting position." },
+      { "id": "genfig05", "text": "A purple unicorn that covers it's mouth with it's hooves.  It's in a sitting position." },
```
[the remark this answers is not in the document]
Comment (snipercup) [reply data/json/items/generic/toys_and_sports.json]: Linted using web linter

File: data/json/mapgen/nested/retail_nested.json [hunk 2]
```diff
@@ -345,6 +345,85 @@
...
+        "  z",
+        "z z"
+      ],
+      "terrain": { "z": "t_floor" },
+      "furniture": { "z": [ [ "f_cardboard_box", 5 ], "f_crate_c" ] },
+      "items": {
+        "z": [
+          { "item": "gift_shop", "chance": 30, "repeat": [ 0, 2 ] }
+        ]
+      }
```
[the remark this answers is not in the document]
Comment (snipercup) [reply data/json/mapgen/nested/retail_nested.json]: Linted using web linter

[the remark this answers is not in the document]
Comment (ashraf1246) [reply data/json/mapgen/nested/retail_nested.json]: @dbackeus @slater

[the remark this answers is not in the document]
Comment (ashraf1246) [reply data/json/mapgen/nested/retail_nested.json]: "id": 4895576,
  "tag": "f38-build-side-60349",
  "tag_id": 60349,
  "task_id": 94461438,
  "event_id": 102710608,
  "with_src": false,
  "with_separate_src": false,
  "with_debuginfo": false
}

**Comment by ashraf1246** [comment]
Fix-LocalStorage-Bugs

File: data/json/items/fluff.json [hunk 3]
```diff
@@ -618,6 +618,24 @@
     "volume": "2100 ml",
     "weight": "745 g"
   },
+  {
+    "id": "incense",
+    "type": "AMMO",
+    "ammo_type": "incense",
+    "category": "other",
+    "name": { "str": "incense", "str_pl": "incense" },
```
Comment (Maleclypse) [inline data/json/items/fluff.json:626]:
```suggestion
    "name": { "str_sp": "incense" },
```

File: data/json/items/generic/toys_and_sports.json [hunk 4]
```diff
@@ -53,6 +53,51 @@
...
+        "id": "genfig05",
+        "text": "A purple unicorn that covers it's mouth with it's hooves.  It's in a sitting position."
+      },
+      { "id": "genfig06", "text": "It seems to wear a reindeer suit.  It has a skull for a head." },
+      { "id": "genfig07", "text": "It looks like a swamp monster with scales over it's body." },
+      { "id": "genfig08", "text": "It looks like the reaper.  In it's left hand is a scythe." },
+      { "id": "genfig09", "text": "A mermaid sitting on a rock.  It has red hair." },
+      {
+        "id": "genfig10",
+        "text": "It seems to wear an avocado suit.  In the left hand is a salt shaker and in the right hand is a spoon. It's in a sitting position."
```
Comment (Maleclypse) [inline data/json/items/generic/toys_and_sports.json:87]:
```suggestion
        "text": "It seems to wear an avocado suit.  In the left hand is a salt shaker and in the right hand is a spoon.  It's in a sitting position."
```

File: data/json/items/generic/toys_and_sports.json [hunk 5]
```diff
@@ -53,6 +53,51 @@
...
+      },
+      {
+        "id": "genfig11",
+        "text": "Looks like someone's hero.  You can bend the arms and legs to give it different stances."
+      }
+    ],
+    "price": 600,
+    "price_postapoc": 20,
+    "material": [ "plastic" ],
+    "weight": "300 g",
```
Comment (Maleclypse) [inline data/json/items/generic/toys_and_sports.json:97]:
```suggestion
    "weight": "100 g",
```

**Pull request merged**